core/blockchain.go

```diff
@@ -694,13 +713,18 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, root common.Hash, repair bo
 	// touching the header chain altogether, unless the freezer is broken
 	if repair {
 		if target, force := updateFn(bc.db, bc.CurrentBlock().Header()); force {
-			bc.hc.SetHead(target, updateFn, delFn)
+			bc.hc.SetHead(target.Number.Uint64(), updateFn, delFn)
 		}
 	} else {
 		// Rewind the chain to the requested head and keep going backwards until a
 		// block with a state is found or fast sync pivot is passed
-		log.Warn("Rewinding blockchain", "target", head)
-		bc.hc.SetHead(head, updateFn, delFn)
+		if head != 0 || time == 0 {
```

Comment:
I'm wondering if it would be better to turn this around (like in `NewBlockchain`)
So setHeadWithTimestamp if time != 0 and setHead otherwise. So a sethead to the genesis block will still be done by number

Reply:
Ah sure, that works